### ISSUE TITLE
Upcoming breakage with transformers-0.6 / mtl-2.3

### ISSUE THREAD

**Issue by andreasabel** [haskell/HTTP#139]
`transformers-0.6` (enabled by upcoming `mtl-2.3`) removes `Control.Monad.Error`. Thus, compilation will break:
```
[ 5 of 18] Compiling Network.Stream   ( Network/Stream.hs, dist/build/Network/Stream.o, dist/build/Network/Stream.dyn_o )

Network/Stream.hs:35:1: warning: [-Wdeprecations]
    Module ‘Control.Monad.Error’ is deprecated:
      Use "Control.Monad.Except" instead
   |
35 | import Control.Monad.Error
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^

Network/Stream.hs:44:10: warning: [-Wdeprecations]
    In the use of type constructor or class ‘Error’
    (imported from Control.Monad.Error, but defined in transformers-0.5.6.2:Control.Monad.Trans.Error):
    Deprecated: "Use Control.Monad.Trans.Except instead"
   |
44 | instance Error ConnError where
   |          ^^^^^
```

**Comment by hsenag** [comment]
Thanks for the heads-up! Is it urgent that this get fixed before or right after the `mtl-2.3` release? The upper bound will correctly exclude it in the short-term but I appreciate that it may be important to get the ecosystem updated quickly.

**Comment by andreasabel** [comment]
I heard that `transformers-0.6` will be part of GHC 9.4, so it would be great if that was released before/with the GHC 9.4 RC.
But I think it would not hurt fixing it now since the deprecation has been out for years.
I can give it a try.

**Comment by hsenag** [comment]
Resolved by #141 - I'll close this once I do a release (planning on this evening, feel free to poke me if I haven't done it in a few days).

**Comment by andreasabel** [comment]
@hsenag : Please also include #142 in the release while we are at it.

**Comment by andreasabel** [comment]
I included a commit in #142 which drops the now obsolete flag `mtl1`.

**Comment by hsenag** [comment]
Now published, thanks for all the cleanups!
